packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -160,6 +161,7 @@ export async function runBuildAndSubmitAsync(projectDir: string, flags: BuildFla
         submitProfile,
         nonInteractive: flags.nonInteractive,
       });
+      startedBuild.build.submission = submission;
```

Comment:
I'm not sure if I like this approach. I think it would be cleaner to refetch the build entity from GraphQL.